.github/workflows/ci.yaml

```diff
@@ -653,3 +640,25 @@ jobs:
           [ "${{matrix.settings.name}}" = "Sqlserver" ] && exit 0
 
           just slt-bin --protocol=flightsql --exclude '*/tunnels/ssh' ${{matrix.settings.path}}
+
+  docker-push:
+    name: Build and Push Docker Image
+    if: github.event_name == 'push' && github.repository == 'GlareDB/glaredb' && github.ref == 'refs/heads/main'
+    runs-on: ubuntu-latest-8-cores
+    permissions:
+      id-token: write # required to use OIDC auth
+      contents: write # required to checkout code
+    env:
+      GCP_PROJECT_ID: glaredb-artifacts
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+      - name: Authenticate to Google Cloud
+        uses: google-github-actions/auth@v2
+        with:
+          credentials_json: ${{ secrets.GCP_SERVICE_ACCOUNT_JSON }}
+      - name: Setup gcloud
+        uses: google-github-actions/setup-gcloud@v2
+      - run: gcloud auth configure-docker -q
+      - run: ./scripts/build-and-push-images.sh
+        # TODO: this should pull the binary that was built in 'build'
```

Comment:
actually this might not be true because we want to use a release build and we build a debug build in test. 

@scsmithr we _could_ make the build we test a release build but then the SLTs wouldn't get overflow checking and similar. 

there might also be less cache reuse.

Reply:
> actually this might not be true because we want to use a release build and we build a debug build in test

fair point, let's remove the comment